### PR TITLE
over/underflow vulnerability updated

### PR DESCRIPTION
**Pull request by 0xSandyy**
## Related Issue

- Closes: https://github.com/kadenzipfel/smart-contract-vulnerabilities/issues/64 

## Checklist

- [x] I have read and followed the [style guide](https://github.com/kadenzipfel/smart-contract-vulnerabilities/blob/master/style-guide.md)

### Describe the changes you've made:
I updated the overflow and underflow vulnerability by adding more ways in which overflow/underflow can occur after solidity >=0.8.

## Type of change

Select the appropriate checkbox:

- [ ] Bug fix (fixing an issue with existing vulnerability data)
- [ ] New feature (adding a new vulnerability or category)
- [x] Documentation update (improving existing information)

## Additional Information

I provided just enough information about different concepts like typecasting, inline assembly, etc without going into detail as this report is mostly about exploiting overflow and underflow. But, there are resources in the source section if readers want to learn more.

### CODE REVIEW

File: vulnerabilities/overflow-underflow.md [hunk 1]
```diff
@@ -1,18 +1,74 @@
...
+// using inline assembly
+}
+
+return result;
+}
+```
+In the above code we are adding ``1`` into the variable with inline assembly and then returning the result. The variable result will overflow and 0 will be returned, despite this the contract will not throw an error or revert.
+
+### Use of unchecked code block:
+Performing arithmetic operations inside the unchecked block saves a lot of gas because it omits several checks and opcodes.But, some of these opcodes are used in default arithmetic operations in 0.8 to check for underflow/ overflow.
```
Comment (kadenzipfel) [inline vulnerabilities/overflow-underflow.md:57]: Missing space before "But" and double space after the following comma

Comment (kadenzipfel) [inline vulnerabilities/overflow-underflow.md:57]: Should remove the space after the slash in "underflow/ overflow"

**Review by kadenzipfel — CHANGES_REQUESTED** [review]
Awesome improvement 🙏

File: vulnerabilities/overflow-underflow.md [hunk 2]
```diff
@@ -1,18 +1,74 @@
...
+The above code snippet will overflow and the ``2`` will be stored in the variable ``b`` due to the fact that maximum value in uint8 data type is ``255``. So, it will overflow and reset to ``0`` without reverting.
 
-### Sources
+### Using Shift Operators
+Overflow & underflow checks are not performed for shift operations like they are performed for other arithmetic operations. Thus, over/underflows can occur.
 
-- https://consensys.github.io/smart-contract-best-practices/attacks/insecure-arithmetic/
+The left shift ``<<`` operator shifts all the beats in the first operand by the number specified in the second operand.Shifting an operand by 1 position is equivalent to multiplying it by 2, shifting 2 positions is equivalent to multiplying it by 4, and shifting 3 positions is equivalent to multiplying by 8.
```
[the remark this answers is not in the document]
Comment (0xSandyy) [reply vulnerabilities/overflow-underflow.md]: Fixed in the latest commit!

File: vulnerabilities/overflow-underflow.md [hunk 3]
```diff
@@ -1,18 +1,74 @@
...
+
+```solidity
+uint8 public a = 255;
+
+function addition() public returns (uint8 result) {
+
+assembly {
+result := add(sload(a.slot),1) // adding 1 will overflow and reset to 0
+// using inline assembly
+}
```
[the remark this answers is not in the document]
Comment (0xSandyy) [reply vulnerabilities/overflow-underflow.md]: Fixed in the latest commit! https://github.com/kadenzipfel/smart-contract-vulnerabilities/commit/e063337001e9783900664a24b537c85e3c49b835

File: vulnerabilities/overflow-underflow.md (hunk 1 above)
Comment (0xSandyy) [reply vulnerabilities/overflow-underflow.md]: Fixed in the latest commit! https://github.com/kadenzipfel/smart-contract-vulnerabilities/commit/e063337001e9783900664a24b537c85e3c49b835

Comment (0xSandyy) [reply vulnerabilities/overflow-underflow.md]: Fixed in the latest commit! https://github.com/kadenzipfel/smart-contract-vulnerabilities/commit/e063337001e9783900664a24b537c85e3c49b835

**Review by kadenzipfel — APPROVED** [review]

**Pull request merged**